### DISCUSSION TITLE
How to use ?

### DISCUSSION THREAD

**Discussion by anhdd-kuro** [takuma-ru/auto-story-generator#5] [Q&A] [answered]
I installed and added settings to main.ts, but after run storybook nothing happened at all.

![スクリーンショット 2024-01-27 15 17 38](https://github.com/takuma-ru/auto-story-generator/assets/13819766/bca9f21e-1fe3-4328-be38-e9088524debd)

**Comment by takuma-ru** [comment]
It is designed to run when it detects a file save.
Try saving `Test.tsx`.

Also, at this time we generate stories based on the type or interface information named `{ComponentName}Props` in the functional components. (This will be adapted to various names in the future.)

**Comment by takuma-ru** [comment]
Idea: It would be nice if the Story file is also generated when `npm run storybook` is run

**Comment by anhdd-kuro** [comment]
Oh I saw it run , but there are some problems what I have found

- Problem 1

If we write props directly

```tsx
export const KTest = ({ text }: { text: string }) => {
  return <div>{text}</div>
}
```

It will throw error

```
Could not find argTypes
```

- Problem 2

If the file name is different from Props type name it also throw error
for example: 

```md
components path : "/components/Foo/index.tsx" 
// →  Could not find argTypes

components : "/components/Foo.tsx" , type Props = {...} ;
export const Foo = ({ ... } : Props) => {
 ...
}

// →  Could not find argTypes

```

- Propblem 3
It also failed for this case when html attributes been passed as props. Look like it has been overwhelmed by the number of props 

```tsx
export type KTestProps = {
  text: string
  count?: number
} & HTMLAttributes<HTMLDivElement>

export const KTest = memo(function MemorizedModal({ text }: KTestProps) {
  return <div>{text}</div>
})
```

May be I should convert to issue ?

**Comment by takuma-ru** [comment]
Thanks for investigating!

Problem 1 and 2 are simply because we don't support the 2 writing styles of that program... 
I know what the solution is, and I'll fix it after I've done some work on the documentation page.

As for Problem 3, I can't figure out the cause just from that code... 
Maybe the error occurred during the conversion to a story after type parsing. However, this writing style is often used, so we will deal with it as soon as possible!

**Comment by takuma-ru** [comment, marked as answer]
Fixed the above problem!

- problem 1: Update program to retrieve even when defined within function arguments
- problem 2: Updated program to recognize directory names as component names when the file name is index.tsx
- problem 3: Updated program to not include HTMLAttributes in arg in story.

**Comment by takuma-ru** [comment]
Documentation will be updated at a later date.
